src/Microsoft.TemplateEngine.Utils/EngineEnvironmentSettings.cs

```diff
@@ -31,6 +31,11 @@ public EngineEnvironmentSettings(ITemplateEngineHost host, Func<IEngineEnvironme
 
         public IPathInfo Paths { get; set; }
 
+        public void Dispose()
+        {
+            (SettingsLoader as IDisposable)?.Dispose();
```

Comment:
consider this
https://docs.microsoft.com/en-us/dotnet/standard/garbage-collection/implementing-dispose

Reply:
I added `if (_disposed)` check.